cmd/juju/firewall/listrules.go

```diff
@@ -76,27 +83,36 @@ func (c *listFirewallRulesCommand) Init(args []string) (err error) {
 // ListFirewallRulesAPI defines the API methods that the list firewall rules command uses.
 type ListFirewallRulesAPI interface {
 	Close() error
-	ListFirewallRules() ([]params.FirewallRule, error)
+	ModelGet() (map[string]interface{}, error)
 }
 
 // Run implements cmd.Command.
 func (c *listFirewallRulesCommand) Run(ctx *cmd.Context) error {
+	ctx.Warningf(deprecationWarning + "\n")
+
 	client, err := c.newAPIFunc()
 	if err != nil {
 		return err
 	}
 	defer client.Close()
-	rulesResult, err := client.ListFirewallRules()
+	cfg, err := client.ModelGet()
 	if err != nil {
 		return err
 	}
+	sshAllow, ok := cfg[config.SSHAllowListKey].(string)
+	if !ok {
+		sshAllow = "0.0.0.0/0"
+	}
```

Comment:
You're dodging the config and validation.

```suggestion
	attrs, err := client.ModelGet()
	if err != nil {
		return err
	}
	cfg, err := config.New(config.NoDefaults, attrs)
	if err != nil {
		return version.Zero, errors.Trace(err)
	}
	sshAllowList := cfg.SSHAllowList()
	if len(list) == 0 {
		....
	}
	....
	
```